_faq/cep-and-its-importance.md

```diff
@@ -0,0 +1,12 @@
+---
+question: "What is a cEP and why it is so important to do analysis pre-gsoc and design as early as possible, to be finalized in the cEP?"
+---
+A **cEP (coala Enhancement Proposal)** is proposed to the coala community discussing the new major features that the contributor intends to implement during GSoC.
+
+**cEP's** are designed before implementation. Implementing major features usually happens from a technical perspective and may lead to poorly designed content - developer and user wise. This can essentially save a lot of coding work when big features are written which are oftentimes not well enough designed initially. This also makes sure that major features are properly documented.
+
+Experience tells us that the best GSoC students are the ones who heavily contributed to the project ideas website, created projects from the issues, and expanded on existing ***.md*** files.
+
+So, if you are planning to do GSoC with coala you should be starting to write detailed project ideas in our [projects](https://github.com/coala/projects) repository. Also, you should learn how to express complex technical ideas in the markup because if you can't explain your idea, and how you will achieve it, we probably don't want to mentor you doing that undefined project.
```

Comment:
You can probably cut `doing that undefined project` from the end of that sentence, since it is already described earlier.